src/simple-notifications/components/notification.component.ts

```diff
@@ -277,26 +277,23 @@ export class NotificationComponent implements OnInit, OnDestroy {
     }
 
     private instance = () => {
-        this.zone.runOutsideAngular(() => {
-            this.zone.run(() => this.diff = (new Date().getTime() - this.start) - (this.count * this.speed));
+        this.diff = (new Date().getTime() - this.start) - (this.count * this.speed);
 
-            if (this.count++ === this.steps) this.zone.run(() => this.remove());
-            else if (!this.stopTime) {
-                if (this.showProgressBar) this.zone.run(() => this.progressWidth += 100 / this.steps);
+        if (this.count++ === this.steps) this.remove();
+        else if (!this.stopTime) {
+            if (this.showProgressBar) this.progressWidth += 100 / this.steps;
 
-                this.timer = setTimeout(this.instance, (this.speed - this.diff));
-            }
-        })
+            this.timer = setTimeout(this.instance, (this.speed - this.diff));
+        }
+        this.changeRef.detectChanges();
     };
 
     private remove() {
         if (this.animate) {
             this.item.state = this.animate + 'Out';
-            this.zone.runOutsideAngular(() => {
```

Comment:
I think you're right I use it a bit excessively, however, I think the setTimeouts should definitely be wrapped?

Reply:
Probably the setTimeouts can be. I'll check that tomorrow :)
I do not like to add a new function on the stack when using OnPush and knowing the change detection will actually not run, but maybe this can be checked at runtime.

Reply:
Just added runOutisde when creating timer, keeps being OK :)